### ISSUE TITLE
SparseLinear and Linear outputs and gradients do not match

### ISSUE THREAD

**Issue by jumutc** [torch/nn#978]
I have spotted small deviations from the ``nn.Linear`` layer while working with a mini batch. In the test example below I have used a small library ``nnsparse`` for sparsification and the output is completely compatible with ``nn.SparseLinear``.

Please consider the following failing test snippet:

```lua
require 'nn'
require 'nnsparse'

local sparseLinear = {}
local tester = torch.Tester()
local sparsifier = function(x) if torch.uniform() < 0.6 then return 0 else return x end end

function sparseLinear.miniBatch() 

	local input = torch.Tensor(10,100):uniform():apply(sparsifier)
	local dloss = torch.Tensor(10, 50):uniform()
	                                
	local sparseInput = input:sparsify()

	local inputSize, outputSize = input:size(2), dloss:size(2)

	local denseLayer  = nn.Linear(inputSize, outputSize)
	local sparseLayer = nn.SparseLinear(inputSize, outputSize)

	-- provide the same weights
	denseLayer.bias   = sparseLayer.bias:clone()
	denseLayer.weight = sparseLayer.weight:clone()

	denseLayer:zeroGradParameters()
	sparseLayer:zeroGradParameters()

	--forward
	local expectedOutput = denseLayer:forward(input)
	local obtainedOutput = sparseLayer:forward(sparseInput)

	tester:assertTensorEq(expectedOutput, obtainedOutput, 1e-6, 'Fail to compute sparseLinear:forward given for a minibatch')

	--backward
	denseLayer:backward (input,       dloss)
	sparseLayer:backward(sparseInput, dloss)

	tester:assertTensorEq(denseLayer.gradBias  , sparseLayer.gradBias  , 1e-6, 'Fail to compute sparseLinear:backward for a minibatch : bias')
	tester:assertTensorEq(denseLayer.gradWeight, sparseLayer.gradWeight, 1e-6, 'Fail to compute sparseLinear:backward for a minibatch : gradWeight')

end

math.randomseed(os.time())

tester:add(sparseLinear)
tester:run()
```

**Comment by jumutc** [comment]
Just installed the latest master and everything works fine... sorry
